lib/web/apiserver.go

```diff
@@ -360,6 +360,8 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 
 	// Database access handlers.
 	h.GET("/webapi/sites/:site/databases", h.WithClusterAuth(h.clusterDatabasesGet))
+	// database token generation
+	h.POST("/webapi/databases/token", h.WithAuth(h.createDatabaseJoinTokenHandle))
```

Comment:
Or should we make an endpoint to retrieve tokens and have the role as parameter?

`POST webapi/token?role=node,app,db`

Reply:
Do you think we could move this to gRPC?

No objections about the combined endpoint on my part, if you think it's feasible.

Reply:
It looks like the only package in the `webapps` project that uses gRPC is `teleterm`.

I've searched docs and asked around but looks like we have no plans to use gRPC outside of `teleterm`, so I think it's best to keep it as a JSON endpoint, what do you think?

Reply:
Yes, I think it makes sense to do it.

But I'm thinking about merging it this way (so we can deliver the feature) and going back and refactoring the endpoints in a future issue if you are not against it.

Does that make sense?

Reply:
Sure, just leave a TODO for yourself (`// TODO(mcbattirola): ...`).